src/Router.php

```diff
@@ -90,24 +90,23 @@ public static function patch(string $path, callable $callback, string|null $midd
         }
     }
 
-    public static function delete(string $path, $callback): void
+    public static function delete($path, $callback, string|null $middleware = null): void
     {
-        if (isset($_REQUEST['_method'])) {
-            if (strtolower($_REQUEST['_method']) !== 'delete') {
-                return;
-            }
-        }
-
         if ($_SERVER['REQUEST_METHOD'] === 'POST') {
-            if ((new self())->getResourceId()) {
-                $path = str_replace('{id}', (string) (new self())->getResourceId(), $path);
+            if ($_POST['_method'] === 'delete') {
+                if ((new self())->getResourceId()) {
+                    $path = str_replace('{id}', (string)(new self())->getResourceId(), $path);
+                    if ($path === parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH)) {
+                        $callback((new self())->getResourceId());
+                        exit();
+                    }
+                }
                 if ($path === parse_url($_SERVER['REQUEST_URI'], PHP_URL_PATH)) {
```

Comment:
There is no need for this block. It can cause unexpected problems. And logically it is impossible delete something without its id